Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/ExploreFeedViewController.m

```diff
@@ -28,51 +29,88 @@ - (void)viewDidLoad {
     self.exploreBasketsTableView.dataSource = self;
     self.exploreBasketsTableView.rowHeight = UITableViewAutomaticDimension;
+    [self getBaskets];
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+    self.navigationController.navigationBar.hidden = YES;
+}
+
+- (void)getBaskets {
     PFQuery *query = [PFQuery queryWithClassName:@"Basket"];
     [query includeKey:@"nonprofits"];
     [query includeKey:@"nonprofits.verificationFiles"];
     [query includeKey:@"createdByUser"];
-    [query findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
+    [query includeKey:@"allTransactions"];
+    [query includeKey:@"featuredValueDict"];
+    [query findObjectsInBackgroundWithBlock:^(NSArray<Basket*> * _Nullable objects, NSError * _Nullable error) {
         if (error != nil) {
             UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Error loading feed" andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
             [self presentViewController:alert animated:YES completion:nil];
         } else {
             self.baskets = objects;
+
+            //FIXME: if object is featured, add basket to featuredBaskets
+            NSArray<Basket*> *sortedFeatured = [objects sortedArrayUsingDescriptors:@[[NSSortDescriptor sortDescriptorWithKey:@"totalFeaturedValue" ascending:NO]]];
+            self.featuredBaskets = [sortedFeatured subarrayWithRange:NSMakeRange(0, MIN(3, sortedFeatured.count))];
             [self.exploreBasketsTableView reloadData];
         }
     }];
 }
 
-- (void)viewWillAppear:(BOOL)animated {
-    self.navigationController.navigationBar.hidden = YES;
-}
-
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     BasketTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"BasketTableViewCell"];
-    Basket *basket = self.baskets[indexPath.row];
+
+    Basket *basket;
+    if (indexPath.section == 0) {
+        basket = self.featuredBaskets[indexPath.row];
+    } else {
+        basket = self.baskets[indexPath.row];
+    }
     cell.basketNameLabel.text = basket.name;
     cell.basketDescriptionLabel.text = basket.basketDescription;
     
     //TODO: show profile pics of nonprofits
+    //FIXME: SHOW BASKET HEADER VIEW, NOT NONPROFITS
     [self getNonprofitImagesFromBasket:basket onCell:cell];
 
     return cell;
 }
 
-//- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
-//
-//}
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+    if (indexPath.section == 0) {
+        self.basketToPass = self.featuredBaskets[indexPath.row];
+    } else {
+        self.basketToPass = self.baskets[indexPath.row];
+    }
+    [self performSegueWithIdentifier:@"showBasketDetail" sender:nil];
+}
+
+/** Sections. */
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    NSInteger numberOfSections = 2;
```

Comment:
I see, I created this var because I wanted to avoid magic numbers as per the code quality guidelines you sent me. What makes a magic number vs what isn't a magic number?

Reply:
Yes, that's perfect, but you should use constants instead of variables. Constants are defined with the `const` keyword, and usually at the class level not the method (or pre-compiler level using `#define` directives).